shopinvader_api_sale_loyalty/schemas/sale.py

```diff
@@ -35,8 +35,11 @@ def from_sale_order(cls, odoo_rec):
             for card in odoo_rec.generated_coupon_ids
         ]
         # Get claimable rewards
-        odoo_rec._update_programs_and_rewards()
         claimable_rewards = odoo_rec._get_claimable_rewards()
+        if not odoo_rec.state == "done" and not claimable_rewards.items():
```

Comment:
After think twice, I think we can simply remove the call to _update_programs_and_rewards

I just test it with this PR : https://github.com/shopinvader/odoo-shopinvader/pull/1552 and the test are green

Reply:
Thank @sebastienbeau for your work, I have included them into this PR (code, description, title). Could you close [this PR](https://github.com/shopinvader/odoo-shopinvader/pull/1552) to make sure all improvements are in one place